test/unit/models/taskBasedStatusUpdate.test.js

```diff
@@ -226,53 +226,104 @@ describe("Update Status based on task update", function () {
   });
 
   describe("Test the Model Function for Changing the status to IDLE based on users list passed", function () {
+    let userId0;
     let userId1;
     let userId2;
     let userId3;
     let userId4;
     let userId5;
+    let userId6;
+    let userId7;
+    let userId8;
+    let userId9;
     let listUsers;
 
     beforeEach(async function () {
       const userArr = userData();
-      userId1 = await addUser(userArr[6]);
-      userId2 = await addUser(userArr[8]);
-      userId3 = await addUser(userArr[9]);
-      userId4 = await addUser(userArr[0]);
-      userId5 = await addUser(userArr[1]);
-      listUsers = [userId1, userId2, userId3, userId4, userId5];
-      await userStatusModel.doc("userStatus001").set(generateStatusDataForState(userId1, userState.ACTIVE));
-      await userStatusModel.doc("userStatus002").set(generateStatusDataForState(userId2, userState.OOO));
-      await userStatusModel.doc("userStatus003").set(generateStatusDataForState(userId3, userState.IDLE));
-      await userStatusModel.doc("userStatus004").set(generateStatusDataForState(userId4, userState.ONBOARDING));
+      userId0 = await addUser(userArr[0]);
+      userId1 = await addUser(userArr[1]);
+      userId2 = await addUser(userArr[2]);
+      userId3 = await addUser(userArr[3]);
+      userId4 = await addUser(userArr[4]);
+      userId5 = await addUser(userArr[5]);
+      userId6 = await addUser(userArr[6]);
+      userId7 = await addUser(userArr[7]);
+      userId8 = await addUser(userArr[8]);
+      userId9 = await addUser(userArr[9]);
+      await userStatusModel.doc("userStatus000").set(generateStatusDataForState(userId0, userState.ACTIVE));
+      await userStatusModel.doc("userStatus001").set(generateStatusDataForState(userId1, userState.OOO));
+      await userStatusModel.doc("userStatus002").set(generateStatusDataForState(userId2, userState.IDLE));
+      await userStatusModel.doc("userStatus003").set(generateStatusDataForState(userId3, userState.ONBOARDING));
+      await userStatusModel.doc("userStatus005").set(generateStatusDataForState(userId5, userState.ACTIVE));
+      await userStatusModel.doc("userStatus006").set(generateStatusDataForState(userId6, userState.OOO));
+      await userStatusModel.doc("userStatus007").set(generateStatusDataForState(userId7, userState.IDLE));
+      await userStatusModel.doc("userStatus008").set(generateStatusDataForState(userId8, userState.ONBOARDING));
+      listUsers = [
+        { userId: userId0, expectedState: "IDLE" },
```

Comment:
what is expected state?

Reply:
is it for asserting against actual response?

Reply:
`expectedState` means based on the tasks they are expected to be in that state. if they have a task assigned they are expected to be active else idle.